datastream-common/src/main/java/com/linkedin/datastream/common/BrooklinEnvelope.java

```diff
@@ -48,11 +51,25 @@ public BrooklinEnvelope(Object key, Object value, Map<String, String> metadata)
    */
   public BrooklinEnvelope(@Nullable Object key, @Nullable Object value, @Nullable Object previousValue,
       Map<String, String> metadata) {
+    this(key, value, previousValue, null, metadata);
+  }
+
+  /**
+   * Construct a {@link BrooklinEnvelope} using record key, value, headers and metadata
+   * @param key The record key (e.g. primary key)
+   * @param value The new record value
+   * @param previousValue The old record value
+   * @param headers Kafka headers to associate with the change event
+   * @param metadata Additional metadata to associate with the change event
+   */
+  public BrooklinEnvelope(@Nullable Object key, @Nullable Object value, @Nullable Object previousValue,
```

Comment:
Just out of curiosity, how are we planning to fit headers within `DatastreamEvent` and `LiDatastreamEvent`?

Reply:
@ahmedahamid we don't need to actually. The ProducerRecord<> takes headers as an argument, so we don't need to modify our own `LiDatastreamEvent` or `DatastreamEvent` AFAIK